nasl/nasl_crypto2.c

```diff
@@ -88,6 +88,70 @@ print_gcrypt_error (lex_ctxt *lexic, char *function, int err)
                gcry_strerror (err));
 }
 
+static int
+find_cipher_hd (cipher_table_item_t *cipher_elem, int *id)
+{
+  if (cipher_elem->id == *id)
+    return 0;
+
+  return -1;
+}
+
+/**
+ * @brief Helper function to validate the cipher id.
+ *
+ * @param[in] cipher_id The cipher ID to validate.
+ * @return Handler on success, Null on error.
+ */
+static gcry_cipher_hd_t
+verify_cipher_id (lex_ctxt *lexic, int cipher_id)
+{
+  cipher_table_item_t *hd;
+  GList *hd_aux;
+
+  hd_aux = g_list_find_custom (cipher_table, &cipher_id,
+                               (GCompareFunc) find_cipher_hd);
+  if (!hd_aux)
+    {
+      nasl_perror (lexic, "Cipher handle %d not found.\n", cipher_id);
```

Comment:
Seems like this should either be done only here or only in the nasl layer.  

encrypt_stream_data is issuing the same error, but nasl_close_stream_cipher is not.